dev_guide/pod_preset.adoc

```diff
@@ -0,0 +1,565 @@
+
+If the pod creation encounters an error, the pod is created without any injected resources from the pod preset.
+//end::pod-preset[]
+
+.Sample pod preset object
+----
+kind: PodPreset
+apiVersion: settings.k8s.io/v1alpha1 <1>
+metadata:
+  name: allow-database <2>
+  namespace: myns
+spec:
+  selector:
+    matchLabels:
+      role: frontend <3>
+  env:
+    - name: DB_PORT <4>
+      value: "6379" <4>
+  envFrom:
+    - configMapRef: <5>
+      name: etcd-env-config
+    - secretKeyRef: <6>
+      name: test-secret
+  volumeMounts: <7>
+    - mountPath: /cache
+      name: cache-volume
+  volumes: <8>
+    - name: cache-volume
+      emptyDir: {}
+----
+
+<1> Specify the `settings.k8s.io/v1alpha1` API.
+<2> Name of the pod preset. This name is used in the pod annotation.
+<3> A label selector that matches the label in the pod specification.
+<4> Creates an environment variable to pass to the container.
+<5> Adds a `ConfigMap` to the pod specification.
+<6> Adds a secrets object to the pod specification.
+<7> Specifies where external storage volumes should be mounted within the container.
+<8> Defines storage volumes that are available to the container(s).
+
+//tag::pod-preset-sample[]
+.Sample pod specification
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: website
+  labels:
+    app: website
+    role: frontend <1>
+  annotations:
+    podpreset.admission.kubernetes.io/exclude: "true" <2>
+spec:
+  containers:
+    - name: website
+      image: ecorp/website
+      ports:
+        - containerPort: 80
+----
+
+<1> A label to match the label selector in the pod preset.
+<2> Annotation to prevent the pod preset from modifying a pod spec.
+
+.Sample pod specification after a pod preset
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: website
+  labels:
+    app: website
+    role: frontend
+  annotations:
+    podpreset.admission.kubernetes.io/allow-database: "resource version" <1>
+spec:
+  containers:
+    - name: website
+      image: ecorp/website
+      volumeMounts: <2>
+        - mountPath: /cache
+          name: cache-volume
+      ports:
+        - containerPort: 80
+      env: <3>
+        - name: DB_PORT
+          value: "6379"
+      envFrom: <4>
+        - configMapRef:
+          name: etcd-env-config
+        - secretKeyRef:
+          name: test-secret
+  volumes: <5>
+    - name: cache-volume
+      emptyDir: {}
+----
+
+<1> The annotation added to show a pod preset was injected, if the pod specification was not configured to prevent the modification.
+<2> The volume mount is added to the pod.
+<3> The environment variable is added to the pod.
+<4> The `ConfigMap` and secrets object added to the pod.
+<5> The volume mount is added to the pod.
+
+
+//end::pod-preset-sample[]
+
+[[dev-guide-pod-presets-create]]
+== Creating Pod Presets
+
+The following example demonstrates how to create and use pod presets.
+
+[[dev-guide-pod-presets-adm]]
+===== Add the Admission Controller
```

Comment:
@mburke5678 I'm wondering if all of the ===== level topics can be merged into one section. They seem to be several steps for the one procedure. Not sure why it jumps from == level to ===== level, too.